apps/moderation/src/components/transparency-log.tsx

```diff
@@ -0,0 +1,293 @@
+
+  const { t } = useTranslation();
+  const listItemObserver = React.useRef<any>();
+
+  React.useEffect(() => {
+    fetchModerationLog();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const anchor = React.useCallback(
+    node => {
+      if (requesting) return;
+      if (listItemObserver.current) listItemObserver.current.disconnect();
+      listItemObserver.current = new IntersectionObserver(entries => {
+        if (entries[0].isIntersecting && nextIndex !== null) {
+          // fetch more entries using nextIndex
+          fetchModerationLog(nextIndex);
+        }
+      });
+      if (node) listItemObserver.current.observe(node);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [requesting],
+  );
+
+  const getStatusCount = async () => {
+    setRequesting(true);
+    try {
+      const response = await moderationRequest.getCount();
+      if (response) {
+        setCount(response);
+      }
+    } catch (error) {
+      logger.error('[transparency-log.tsx]: getStatusCount err %j', error.message || '');
+    } finally {
+      setRequesting(false);
+    }
+  };
+
+  const fetchModerationLog = async (offset?: string, limit?: number) => {
+    // fetch log of moderated contents
+    setRequesting(true);
+    try {
+      const response = await moderationRequest.getLog({
+        offset,
+        limit: limit || DEFAULT_LIMIT,
+      });
+      getStatusCount();
+      if (response.results) {
+        setLogItems([...logItems, ...response.results]);
+      }
+      setNextIndex(response.nextIndex);
+    } catch (error) {
+      logger.error('[transparency-log.tsx]: fetchModerationLog err %j', error.message || '');
+    } finally {
+      setRequesting(false);
+    }
+  };
+
+  const onTabClick = (value: string) => {
+    setActiveButton(buttonValues[buttonLabels.indexOf(value)]);
+  };
+
+  const handleClickArrowLeft = () => {
+    setSelected(null);
+  };
+
+  const handleClickViewItem = (contentType: string, contentID: string) => () => {
+    if (contentType === 'post') {
+      navigateToUrl(`${BASE_SOCIAL_URL}/post/${contentID}`);
+    } else if (contentType === 'reply' || contentType === 'comment') {
+      navigateToUrl(`${BASE_SOCIAL_URL}/reply/${contentID}`);
+    } else if (contentType === 'account') {
+      navigateToUrl(`${BASE_PROFILE_URL}/${contentID}`);
+    }
+  };
+
+  const handleClickAvatar = () => {
+    /* TODO */
+  };
+
+  const handleClickContactModerators = () => {
+    /* TODO */
+  };
+
+  const handleClickCard = (el: ILogItem) => () => {
+    selected?.contentID === el.contentID ? setSelected(null) : setSelected(el);
+  };
+
+  const buttonValues = !isMobile
+    ? ['All', 'Kept', 'Delisted']
+    : ['All', 'Kept', 'Delisted', 'Stats'];
+  const buttonLabels = !isMobile
+    ? [t('All'), t('Kept'), t('Delisted')]
+    : [t('All'), t('Kept'), t('Delisted'), t('Stats')];
+
+  const baseAvatarUrl = 'https://hub.textile.io/ipfs/';
```

Comment:
Why is this defined as a constant here? this url should come from the sdk, like: sdk.services.common.ipfs.getSettings().gateway

Reply:
gotcha 👍